### PR TITLE
AC-817 Create FormEntity and update other entity classes

### PR DESCRIPTION
**Pull request by rishabh-997**
## Description of what I changed
1. Added formEntity to replace ActiveAndroid.
2. Looked through entire code to create queries in FormDAO interface
3. Updated Patient and Encounter Entities
4. Added new and updated previous tests depending on entity

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-817

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Query("SELECT * FROM forms WHERE uuid = :uuid")
+    Single<FormResourceEntity> getFormByUuid(String uuid);
+
+    @Query("DELETE FROM forms")
+    void deleteALlForms();
+
+    @Insert
+    void saveFormResource(FormResourceEntity formResourceEntity);
+}
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java:43]: addFormResource will be more apt !

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java [hunk 2]
```diff
@@ -45,6 +45,8 @@
     private String country;
     @ColumnInfo(name = "postalCode")
     private String postalCode;
+    @ColumnInfo(name = "dead")
+    private String deceased;
     @Embedded(prefix = "patient_")
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java:50]: @rishabh-997 maybe this column name and param name can be made same and something which is more descriptive than simply dead??

File: openmrs-client/src/androidTest/java/org/openmrs/mobile/listeners/watcher/dao/PatientRoomDAOTest.java [hunk 3]
```diff
@@ -161,7 +163,7 @@ private PatientEntity newPatientEntity(long id, String uuid, String display,
                                            String country, String deathDate, EncounterEntity encounterEntity,
                                            String familyName, String gender, String givenName,
                                            String identifier, String middleName, String photo,
-                                           String postalCode, String state, boolean synced) {
+                                           String postalCode, String state, boolean synced, String dead) {
         PatientEntity entity = new PatientEntity();
```
Comment (LuGO0) [inline openmrs-client/src/androidTest/java/org/openmrs/mobile/listeners/watcher/dao/PatientRoomDAOTest.java:167]: if you update that in the entity update it here as well

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java (hunk 2 above)
Comment (rishabh-997) [reply openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java:50]: the column name is the same as that in the server, so I preferred not to change it.

File: openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java (hunk 1 above)
Comment (rishabh-997) [reply openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java:43]: Actually all other DAO-interfaces uses `saveTable`  naming, so I tried to make similar type of functions have similar naming. I will need to change all the DAO-interfaces with the same in that case

Comment (LuGO0) [reply openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java:43]: ohkk @rishabh-997 :+1

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java (hunk 2 above)
Comment (LuGO0) [reply openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java:50]: yes @rishabh-997 I was asking if we can we make the param same to it??

File: openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java (hunk 1 above)
Comment (f4ww4z) [reply openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java:43]: > Actually all other DAO-interfaces uses `saveTable` naming, so I tried to make similar type of functions have similar naming. I will need to change all the DAO-interfaces with the same in that case

@rishabh-997 This is better, can you do this right in this PR?

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/EncounterEntity.java [hunk 4]
```diff
@@ -19,6 +19,10 @@
     private String patientUuid;
     @ColumnInfo(name = "form_uuid")
     private String formUuid;
+    @ColumnInfo(name = "location_uuid")
+    private String locationUuid;
+    @ColumnInfo(name = "encounterProviders_uuid")
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/EncounterEntity.java:24]:
```suggestion
    @ColumnInfo(name = "encounter_provider_uuid")
```

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java (hunk 2 above)
Comment (f4ww4z) [reply openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java:50]: @rishabh-997 @LuGO0 I think we can leave it as it is for now, we can change it later.

File: openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java (hunk 1 above)
Comment (rishabh-997) [reply openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java:43]: yes. I have updated all the `saveEntity` to `addEntity`

**Comment by rishabh-997** [comment]
I have updated PR with requested changes and now merging this PR. I have updated all instances of LocationDAO to LocationRoomDAO and I **need the changes in this PR  to proceed further** with other dao's. I do not have admin rights on JIRA so please close the issue there. If you need me to address more changes here, please add the suggestion and I will incorporate them in the final migration @f4ww4z si :)

**Pull request merged**